firmware/quadruna/VC/src/app/states/app_allStates.c

```diff
@@ -21,4 +30,38 @@ void app_allStates_runOnTick100Hz(void)
     app_heartbeatMonitor_checkIn();
     app_heartbeatMonitor_tick();
     app_heartbeatMonitor_broadcastFaults();
+
+    if (num_cycles <= IGNORE_HEARTBEAT_CYCLES)
+    {
+        num_cycles++;
+    }
+
+    app_heartbeatMonitor_checkIn();
+    app_heartbeatMonitor_tick();
+
+    if (num_cycles > IGNORE_HEARTBEAT_CYCLES)
+    {
+        app_heartbeatMonitor_broadcastFaults();
+    }
+
+    const bool left_inverter_fault  = app_canRx_INVL_VsmState_get() == INVERTER_VSM_BLINK_FAULT_CODE_STATE;
+    const bool right_inverter_fault = app_canRx_INVR_VsmState_get() == INVERTER_VSM_BLINK_FAULT_CODE_STATE;
+    app_canAlerts_VC_Fault_LeftInverterFault_set(left_inverter_fault);
+    app_canAlerts_VC_Fault_RightInverterFault_set(right_inverter_fault);
+
+    const bool bms_fault  = app_canAlerts_BoardHasFault(BMS_ALERT_BOARD);
+    const bool vc_fault   = app_canAlerts_BoardHasFault(VC_ALERT_BOARD);
+    const bool fsm_fault  = app_canAlerts_BoardHasFault(FSM_ALERT_BOARD);
+    const bool crit_fault = app_canAlerts_BoardHasFault(CRIT_ALERT_BOARD);
+
+    // once crit dim is done add dim_fault
+    const bool any_board_has_fault = bms_fault || vc_fault || fsm_fault || crit_fault;
+
+    io_sbgEllipse_handleLogs();
+    app_sbgEllipse_broadcast();
+
+    // Set status to false (which blocks drive) if either inverter is faulted, or another board has set a fault.
+    return !(left_inverter_fault || right_inverter_fault || any_board_has_fault);
```

Comment:
one day everything that happens here gets moved to a diff app file, but for now it is ok.